src/helpers/MiscFunctions.cpp

```diff
@@ -0,0 +1,20 @@
+#include <filesystem>
+#include "MiscFunctions.hpp"
+
+std::string absolutePath(const std::string& rawpath, const std::string& rawcurrentpath) {
+    std::filesystem::path path(rawpath);
+
+    // Handling the case where rawpath starts with '~'
+    if (!rawpath.empty() && rawpath[0] == '~') {
+        static const char* const ENVHOME = getenv("HOME");
+        return std::filesystem::path(ENVHOME) / path.relative_path().string().substr(2);
+    }
+    // Handling e.g. ./, ../
+    else if (path.is_relative()) {
+        const std::filesystem::path currentDir = std::filesystem::path(rawcurrentpath).parent_path();
+
+        return std::filesystem::canonical(currentDir / path);
```

Comment:
this can throw if the thing couldn't be found. We should return an empty string if tomething throws IMO or return an `std::optional` (I'd prefer `std::expected` but clang will shit itself)

Reply:
Okay, I've given my best, but I'm a bit confused if some of these error cases can even occur in handleSource. However, due to absolutePaths optionality, I needed to handle it anyway. I tested multiple configurations (invalid paths, etc.), and it should work now